### PR TITLE
[translation] support translation kwargs for single translation overload

### PR DESCRIPTION
**Pull request by kristapratico**
Resolves https://github.com/Azure/azure-sdk-for-python/issues/19798

https://apiview.dev/Assemblies/Review/019355482c6245a5a404eef64740da1e?diffRevisionId=39de9ee01595481abdf9749b09966197&doc=False&diffOnly=True

### CODE REVIEW

File: sdk/translation/azure-ai-translation-document/azure/ai/translation/document/_client.py [hunk 1]
```diff
@@ -131,6 +131,23 @@ def begin_translation(
...
+            sub folders for translation.
+        :keyword str suffix: A case-sensitive suffix string to filter documents in the source path for
+            translation. This is most often use for file extensions.
+        :keyword storage_type: Storage type of the input documents source string. Possible values
+            include: "Folder", "File".
+        :paramtype storage_type: str or ~azure.ai.translation.document.StorageInputType
+        :keyword str category_id: Category / custom model ID for using custom translation.
+        :keyword glossaries: Glossaries to apply to translation.
+        :paramtype glossaries: list[~azure.ai.translation.document.TranslationGlossary]
+        :keyword str source_storage_source: Source storage source. Default value: "AzureBlob".
```
Comment (iscai-msft) [inline sdk/translation/azure-ai-translation-document/azure/ai/translation/document/_client.py:147]: Slightly weird how you have "Default value is "AzureBlob". Only "AzureBlob" supported"

I think for now you can just have the "Only AzureBlob supported" for now, and once more values get added, switch it to "Default value is AzureBlob"

File: sdk/translation/azure-ai-translation-document/azure/ai/translation/document/_helpers.py [hunk 2]
```diff
@@ -44,14 +45,39 @@ def get_translation_input(args, kwargs, continuation_token):
             target_language_code = kwargs.pop("target_language_code", None)
             if not target_language_code:
                 target_language_code = args[2]
+
+            # Additional kwargs
+            source_language_code = kwargs.pop("source_language_code", None)
+            prefix = kwargs.pop("prefix", None)
+            suffix = kwargs.pop("suffix", None)
+            storage_type = kwargs.pop("storage_type", None)
+            source_storage_source = kwargs.pop("source_storage_source", None)
```
Comment (iscai-msft) [inline sdk/translation/azure-ai-translation-document/azure/ai/translation/document/_helpers.py:54]: just to make sure: are you getting the "AzureBlob" default from the generated code / service, and that's why you can default to None here?

**Review by iscai-msft — APPROVED** [review]

File: sdk/translation/azure-ai-translation-document/azure/ai/translation/document/_client.py (hunk 1 above)
Comment (maririos) [reply sdk/translation/azure-ai-translation-document/azure/ai/translation/document/_client.py:147]: curious, why is Python exposing this property if the user can't set other values right now?
in .NET we decided to hide it, and whenever there are other options we will expose it.

**Review by maririos — APPROVED** [review]

File: sdk/translation/azure-ai-translation-document/azure/ai/translation/document/_client.py (hunk 1 above)
Comment (kristapratico) [reply sdk/translation/azure-ai-translation-document/azure/ai/translation/document/_client.py:147]: I was on the fence about it and Johan didn't have a preference for exposing it. Seeing as .NET is hiding it, I will hide as well. Thanks!

File: sdk/translation/azure-ai-translation-document/azure/ai/translation/document/_helpers.py (hunk 2 above)
Comment (kristapratico) [reply sdk/translation/azure-ai-translation-document/azure/ai/translation/document/_helpers.py:54]: yes service defaults to AzureBlob

**Pull request merged**